psycopg/psycopg/cursor.py

```diff
@@ -275,6 +287,9 @@ def _stream_send_gen(
         self._last_query = query
 
     def _stream_fetchone_gen(self, first: bool) -> PQGen[Optional["PGresult"]]:
+        assert (
+            not self._conn._pipeline_mode
+        ), "pipeline mode not supported"  # TODO
```

Comment:
I would just deny the use of `stream()` in pipeline mode. They seem two different use cases:
- pipeline is to send several operations in parallel from separate threads
- stream is for a single slow operation that doesn't produce all the result at the time of call.
So I don't see why people should try to call stream() in a pipeline() block :)

So I would just make calling `stream()` under `pipeline()` to raise a `ProgrammingError`.